src/browser/lib.js

```diff
@@ -522,18 +524,44 @@ var ASYNC_SAFE = false;
             }
             return;
         }
-
-        const part_filename = this.basename + "-" + offset + "-" + (offset + len) + this.extension;
-
-        v86util.load_file(part_filename, {
-            done: function done(buffer)
-            {
-                dbg_assert(buffer.byteLength === len);
-                var block = new Uint8Array(buffer);
-                this.handle_read(offset, len, block);
-                fn(block);
-            }.bind(this),
-        });
+		
+		if(this.use_step_) {
+			const fake_offset = parseInt(offset / this.step_, undefined) * this.step_;
+			const m_offset = offset - fake_offset;
+			const total_count = parseInt(len / this.step_, undefined) + 2;
+			var blocks_ = new Uint8Array(m_offset + (total_count * this.step_));
+			var finished = 0;
+			
+			for (var i = 0; i < total_count; i++) {
+				const cur = i * this.step_;
+				const part_filename = this.basename + "-" + (cur + fake_offset) + this.extension;
+				
+				v86util.load_file(part_filename, {
+					done: function done(buffer) {
+						const block = new Uint8Array(buffer);
+						blocks_.set(block, cur);
+						const tmp_blocks = blocks_.slice(m_offset, m_offset + len);
+						finished++;
+						if (finished == total_count) {
+							fn(tmp_blocks);
+						}
+					}.bind(this),
+				});
+			}		
+		}
+		else {
+			const part_filename = this.basename + "-" + offset + "-" + (offset + len) + this.extension;
+
+			v86util.load_file(part_filename, {
+				done: function done(buffer)
+				{
+					dbg_assert(buffer.byteLength === len);
+					var block = new Uint8Array(buffer);
+					this.handle_read(offset, len, block);
+					fn(block);
+				}.bind(this),
+			});
+		}
```

Comment:
Please follow the style of the rest of the code:

- No space after `if`
- `{` on its own line
- Indent with 4 spaces
- double quotes
- semicolon at end of line
- `===` over `==`
- No `_` at the end of properties